indigo_app/forms.py

```diff
@@ -21,29 +20,31 @@ class Meta:
 
     publication_document_file = forms.FileField(required=False)
 
-    # not as simple as this
-    # if there's currently nothing there, upload
-    # if there is, replace
     # also check for hidden field to delete
     def save(self, commit=True):
-        work = super(WorkForm, self).save()
         self.save_publication_document()
+        work = super(WorkForm, self).save(commit)
         return work
 
     def save_publication_document(self):
         pub_doc_file = self.cleaned_data['publication_document_file']
         if pub_doc_file:
-
-            pub_doc = PublicationDocument(work=self.instance, file=pub_doc_file,
-                                          size=pub_doc_file.size, filename=pub_doc_file.name,
-                                          mime_type="application/pdf")
-            try:
+            # if one already exists for this work, update it
+            if hasattr(self.instance, 'publication_document'):
```

Comment:
You want to check `if self.instance.publication_document`. This is because `hasattr(x, 'foo')` will be True even if `x.foo` is None.

Reply:
`hasattr(x, 'foo')` is usually used before calling `x.foo` if you're unsure if `x` actually has an attribute `foo`. If it doesn't, then `x.foo` would raise an exception